### ISSUE TITLE
Add currency converter

### ISSUE THREAD

**Issue by israelst** [turicas/rows#32]
Convert a string made of a number and a [currency symbol](https://en.wikipedia.org/wiki/Currency_symbol#List_of_presently-circulating_currency_symbols) to float or to a Currency Type.

I think that we should write a simple version of this converter supporting a few frequently used currency, such as:

 * [ ] Dolar
 * [ ] Euro
 * [ ] Yen
 * [ ] Real
 * [ ] Bitcoin

And then, eventually, build it seriously as another project.

What do you think?

**Comment by turicas** [comment]
Good issue, @israelst! Currently I'm dealing with this kind o problem manually parsing to `rows.fields.DecimalField` (see the [api-rewrite branch](https://github.com/turicas/rows/tree/api-rewrite)). I think we should implment this as a class that inherits from `rows.field.Field` (`rows.fields.CurrencyField`).
We may try to parse the desired string using a regexp based on `locale.localeconv()['currency_symbol']` and/or `locale.localeconv()['int_curr_symbol']`. Or it may be easier for the user if she can specify the regexp directly or the currency name (BRL, USD etc.), since the user may want to convert some data in a specified locale (or the default `C` locale) but only the currency column in this specific locale.
We may use [babel](http://babel.pocoo.org/) for handling this kind of operation.

**Comment by turicas** [comment]
This is not something I can work right now since there are other issues with higher priority but I think before version 1.0.0. we need support for automatically handling currencies. ;-)
Any suggestion of Python library to help dealing with this is welcome. =)

**Comment by turicas** [comment]
For values such as `R$ 2.048,12` you can use this simple class:

```python
from decimal import Decimal

import rows


class PtBrCurrencyField(rows.fields.DecimalField):

    @classmethod
    def deserialize(cls, value, *args, **kwargs):
        if rows.utils.is_null(value):
            return None

        value = value.replace('R$', '').replace('.', '').replace(',', '.')\
                     .strip()
        return Decimal(value)
```

The value will be represented as `decimal.Decimal` (note that you'll loose currency (`R$`) information).

**Comment by mschonfinkel** [comment]
Don't [money](https://github.com/carlospalol/money) or [python-money](https://github.com/poswald/python-money) already do that? Or are these solutions too overkill?

**Comment by turicas** [comment]
@mschonfinkel, good point! I think the best implementation for this field in rows would be to use one of these libraries. :)
